### PR TITLE
feat: hatchling build system, update workflows

### PR DESCRIPTION
**Pull request by manzt**
## Description

What was changed in this pull request?

- Migrates `clodius` to a `hatching` build system (pyproject.toml-only, no `setup.py` or `setup.cfg`)
- Updates GH Actions with separate workflows for linting/testing
- GH Action caches `data/*` (uses hash of `.gitignore` and `get_test_data.sh` as cache key)
- Pins cooler & numpy versions for compatability
- Consolidate flake8 config to just `.flake8`. 

Why is it necessary?

Some much needed updates to use clodius in modern projects.

Fixes #\_\_\_

## Checklist

-   [x] Unit tests added or updated
-   [ ] Updated CHANGELOG.md
-   [ ] Run `black .`

### CODE REVIEW

**Comment by manzt** [comment]
@nvictus pybbi didn't compile with 3.11 so I haven't included

**Comment by manzt** [comment]
Ready for review...

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Cache Fixtures
+      id: cache-fixtures
+      uses: actions/cache@v3
+      with:
+        path: data/
+        key: ${{ runner.os }}-{{ hashFiles('get_test_data.sh') }}-{{ hashFiles('.gitignore') }}
```
Comment (manzt) [inline .github/workflows/ci.yml:42]: creates a key from the hash to `get_test_data.sh` and `.gitignore`. Cache is invalidated if either of these change.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+[project]
+name = "clodius"
+version = "0.19.0"
+description = "Tile generation for big data"
+authors = [
+    { name = "Peter Kerpedjiev", email = "pkerpedjiev@gmail.com" },
+]
+dependencies = [
+    "numpy<=1.23.0",
```
Comment (manzt) [inline pyproject.toml]: cooler < 9.0 relies on `np.object` which has been deprecated

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    "negspy",
+    "pysam",
+    "dask",
+    "requests",
+    "h5py>=3.0.0",
+    "pandas>=1.0",
+    "slugid",
+    "sortedcontainers",
+    "nose",
+    "cooler>=0.8.5, <0.9.0",
```
Comment (manzt) [inline pyproject.toml]: `tiles/cooler.py` is not compatible with `cooler` v0.9

**Comment by manzt** [comment]
it might be worth thinking about breaking about the dependencies of clodius so modules can be installed separately in the future. e.g.,

```
pip install clodius[all]
```

vs.,

```
pip install clodius[cooler,bigbed]
```

**Comment by manzt** [comment]
The linting error yields real errors now for undeclared variables. not sure how to resolve.

**Comment by manzt** [comment]
I've opened another PR on top of this one to to migrate to Ruff (from flake8) and actually check for formatting/linting in CI: https://github.com/manzt/clodius/pull/2

**Comment by nvictus** [comment]
> The linting error yields real errors now for undeclared variables. not sure how to resolve.

It's a broken function. Let's fix or remove it.

**Pull request merged**